lib/navigation/crosscheck.hpp

```diff
@@ -0,0 +1,47 @@
+#include "consts.hpp"
+
+#include <array>
+
+#include "core/types.hpp"
+
+namespace hyped::navigation {
+
+class Crosscheck {
+ public:
+  Crosscheck();
+
+  /**
+   * @brief checks that all sensors agree with wheel encoders. This is a safety
+   * check for the reliability of the current trajectory (i.e. the sensors are
+   * probably not all wrong...)
+   *
+   * @return true signifies trajectory agreement
+   * @return false signifies trajectory disagreement. We enter fail state
+   */
+  SensorChecks checkTrajectoryAgreement(const core::ImuData imu_data,
```

Comment:
Integration functionality of some format will probably be called in here but is still up for debate. Not 100% sure what to pass in here but currently passing in preprocessed data. What would probably make sense at some point is to define a struct for sensor estimates for displacement somewhere and update that with integrated values. However this is further down todo list, really just want basic outline merged for now because this pr is starting to bottleneck nav progress

Reply:
You can open PRs into this branch; there's no rush in getting this merged. (The base will change once this is merge so people can start work on the areas that seem stable.)

Implementation is the easy bit and this is the hard bit that you should get right. It's worth taking time to get a good interface done because then the implementation will write itself.

Reply:
@maxguy2001 if this is the case, we could look at implementing now instead of waiting for merge? We'll probably be able to work out some of these kinks as we implement the other more stable parts of the skeleton.

Reply:
@maxguy2001 if this is the case, we could look at implementing now instead of waiting for merge? We'll probably be able to work out some of these kinks as we implement the other more stable parts of the skeleton.